integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java

```diff
@@ -234,30 +219,26 @@ public TBase create() {
 
       wrapperTableInfo.setTransactionalTable(isTransactionalTable);
 
+      CarbonMetadata.getInstance().removeTable(wrapperTableInfo.getTableUniqueName());
       // Step 4: Load metadata info into CarbonMetadata
       CarbonMetadata.getInstance().loadTableMetadata(wrapperTableInfo);
 
-      cache.carbonTable =
-          CarbonMetadata.getInstance().getCarbonTable(table.getSchemaName(), table.getTableName());
-
-      // cache the table
-      carbonCache.get().put(table, cache);
+      cache.setCarbonTable(
+          CarbonMetadata.getInstance().getCarbonTable(table.getSchemaName(), table.getTableName()));
```

Comment:
Does it need to consider that CarbonMetadata.getInstance().getCarbonTable(table.getSchemaName(), table.getTableName()) return null ？

Reply:
It is not supposed be null as in above line only the table is loaded. But added the null check now.